gcce.service.js

```diff
@@ -102,7 +102,9 @@ module.exports = class GoogleComputeService extends Compute {
         : saAccessScopes
     );
     const config = removeUndefinedAndEmpty({
-      machineType: machineType ? `projects/${this.projectId}/zones/${resolvedZone}/machineTypes/${machineType}` : undefined,
+      ...(machineType ? {
```

Comment:
These conditional destructs are a neat way to achieve the desired functionality in a single line of code, but it's not a common approach, so it will cause some cognitive load on a person new to the problem.
Please replace all of these with simple conditional statements. It will occupy slightly more space, but will be much more readable.